i3lock.c

```diff
@@ -658,8 +659,18 @@ static void raise_loop(xcb_window_t window) {
     }
 }
 
+char* whoami(void) {
+    uid_t uid = geteuid();
+    struct passwd *pw = getpwuid(uid);
+    if (pw) {
+        return pw->pw_name;
+     } else {
+        errx(EXIT_FAILURE, "Username not known!\n");
```

Comment:
Make this `err(EXIT_FAILURE, "getpwuid() failed");`. See the manpage err(3) — `err()` will describe `errno`, which `getpwuid()` sets on failure.